src/AppInstallerCommonCore/Public/winget/Registry.h

```diff
@@ -251,6 +251,12 @@ namespace AppInstaller::Registry
         std::optional<Key> SubKey(std::string_view name, DWORD options = 0) const;
         std::optional<Key> SubKey(const std::wstring& name, DWORD options = 0) const;
 
+        template <typename T>
+        bool SetKeyValue(const std::wstring& name, const std::wstring& value, T dwType)
+        {
+            return TrySetRegistryValue(m_key, name, value, dwType);

```

Comment:
There is not often need to pass member variables to other member functions, and I don't see that it is needed here.

Reply:
Removed `TrySetRegistryValue()` function and call RegSetValuExW() directly